### PR TITLE
Add support for GCS emulators

### LINKED ISSUE #176: Support using GCS emulators
Official Google Cloud Storage libraries usually allow configuring them to perform requests against a GCS emulator server, instead of the real service API.

The user can define a `STORAGE_EMULATOR_HOST` environment variable pointing to a emulator server. When defined, client libraries will:
* Direct GCS API requests to that host
* Disable (or don't enforce) authentication

It would be great to have support for GCS emulators in `googleCloudStorageR` as well. This is generally useful in the context of automated tests. And depending on the emulator implementation, it can even give local filesystem support to applications that were previously hardwired to use GCS.

I've already sketched a solution that I have been using for my own purposes in the last few days. I'll open a PR shortly.

### PR DESCRIPTION
**Pull request by manuteleco**
Allow configuring a Google Cloud Storage emulator to be used instead of the real service. This can be achieved by providing a `STORAGE_EMULATOR_HOST` environment variable with the scheme, host and port of the emulator server (e.g.: http://localhost:8080).

Besides using a different API endpoint, configuring an emulator also has the effect of lifting the requirement of providing valid authentication credentials. Emulators don't require or even implement authentication.

This behavior is similar to the one implemented in official Google Cloud Storage client libraries developed and maintained by Google.

---

Closes #176

It also has a weak dependency with https://github.com/MarkEdmondson1234/googleAuthR/pull/228. Without that PR in `googleAuthR`, using an emulator will still work in `googleCloudStorageR` but it will unfortunately require valid credentials for the real GCS.

---

Just like in https://github.com/cloudyr/googleCloudStorageR/pull/175#issuecomment-1646794932, I wonder whether I have to build the `pkgdown` documentation here or not. I also wonder what else should I include in this PR (e.g., an entry in `NEWS.md`, etc.).

Also, be aware that I'm new to R, so feel free to point out any unidiomatic or plain odd code that needs to be fixed.

---

Finally, for testing I have used https://github.com/fsouza/fake-gcs-server, but it currently requires a few adjustments for it to work with `googleCloudStorageR`: https://github.com/fsouza/fake-gcs-server/pull/1258 and https://github.com/fsouza/fake-gcs-server/pull/1252. With some luck, they will be accepted and merged eventually.

I did not test every single GCS operation; it is possible some are still not working with an emulator yet. But I believe this PR already brings value even if it doesn't happen to provide 100% support. Any operation that is still not working could be refined later, once encountered.

### CODE REVIEW

File: man/gcs_upload.Rd [hunk 1]
```diff
@@ -18,7 +18,7 @@ gcs_upload(
   upload_type = c("simple", "resumable")
 )
 
-gcs_upload_set_limit(upload_limit = 5000000L)
+gcs_upload_set_limit(upload_limit = 5e+06)
```
Comment (manuteleco) [inline man/gcs_upload.Rd:21]: I find this annoying. Not sure why roxygen2 decided to make this change.

**Comment by MarkEdmondson1234** [comment]
Please do add a note to the NEWS.md and feel free to add as a contributor to DESCRIPTION - thank you!  Looks like a useful addition.

**Comment by manuteleco** [comment]
Just in case you didn't get a notification from my last change, I have adjusted `NEWS.md` and `DESCRIPTION`. Let me know if there is anything else I should do :+1: .

**Comment by MarkEdmondson1234** [comment]
@manuteleco great thank you!  I was on holiday

**Comment by MarkEdmondson1234** [comment]
@manuteleco I am getting some bug reports since this was merged, is it possible this is related?

https://github.com/cloudyr/googleCloudStorageR/issues/183

**Pull request merged**